source/pervasive/nonlinear_arith/div.rs

```diff
@@ -0,0 +1,1088 @@
+//! Lemmas for division.
+
+use builtin::*;
+use builtin_macros::*;
+#[allow(unused_imports)]
+use crate::calc_macro::*;
+
+verus! {
+
+#[allow(unused_imports)]
+use crate::nonlinear_arith::internals::div_internals::{div_recursive, lemma_div_induction_auto, div_auto, div_pos, lemma_div_auto};
+#[allow(unused_imports)]
+use crate::nonlinear_arith::internals::div_internals_nonlinear as DivINL;
+#[allow(unused_imports)]
+use crate::nonlinear_arith::internals::mod_internals::{lemma_mod_auto, mod_recursive};
+#[allow(unused_imports)]
+use crate::nonlinear_arith::internals::mod_internals_nonlinear as ModINL;
+#[allow(unused_imports)]
+use crate::nonlinear_arith::internals::mul_internals::{lemma_mul_induction_auto, lemma_mul_auto, mul_auto, lemma_mul_induction};
+#[allow(unused_imports)]
+use crate::nonlinear_arith::mul::*;
+#[allow(unused_imports)]
+use crate::nonlinear_arith::internals::general_internals::{is_le};
+#[allow(unused_imports)]
+use crate::nonlinear_arith::modulus::*;
+
+/*****************************************************************************
+ * Division:
+ *****************************************************************************/
+
+/// the common syntax of division gives the same quotient as performing division through recursion
+// #[verifier::spinoff_prover]
+pub proof fn lemma_div_is_div_recursive(x: int, d: int)
+    requires 0 < d
+    ensures div_recursive(x, d) == x / d
+{
+    reveal(div_recursive);
+    reveal(div_pos);
+    lemma_div_induction_auto(d, x, |u: int| div_recursive(u, d) == u / d);
+}
+
+// #[verifier::spinoff_prover]
+pub proof fn lemma_div_is_div_recursive_auto()
+    ensures forall |x: int, d: int| d > 0 ==> div_recursive(x, d) == #[trigger](x / d)
+{
+    reveal(div_recursive);
+    assert forall |x: int, d: int| d > 0 implies div_recursive(x, d) == #[trigger](x / d) by
+    {
+    lemma_div_is_div_recursive(x, d);
+    }
+}
+
+/// the quotient of an integer divided by itself is 1
+// #[verifier::spinoff_prover] 
+pub proof fn lemma_div_by_self(d: int)
+    requires d != 0
+    ensures d / d == 1
+{
+    DivINL::lemma_div_by_self(d);
+}
+
+/// zero divided by an integer besides 0 is 0
+// #[verifier::spinoff_prover]
+pub proof fn lemma_div_of0(d: int)
+    requires d != 0
+    ensures 0 as int  / d == 0
+{
+    DivINL::lemma_div_of0(d);
+}
+
+/// ensures the basic propoerties of division: 0 divided by any integer is 0; any integer 
```

Comment:
Typo `propoerties` -> `properties`

Also a more useful doc comment is probably something like "A convenience function to introduce basic properties about division"?